app/src/main/java/com/eveningoutpost/dexdrip/watch/miband/MiBandService.java

```diff
@@ -253,114 +347,105 @@ public int onStartCommand(Intent intent, int flags, int startId) {
 
     private void handleCommand() {
         if (messageQueue.isEmpty()) return;
-        queueItem = messageQueue.poll();
+        do {
+            queueItem = messageQueue.poll();
+        } while (queueItem.isExpired() && !messageQueue.isEmpty());
+        if (queueItem.isExpired()) return;
+
         switch (queueItem.functionName) {
             case "refresh":
                 whenToRetryNextBgTimer(); //recalculate isNightMode
                 ((MiBandState) mState).setSettingsSequence();
-                changeState(INIT);
                 break;
             case "message":
                 ((MiBandState) mState).setQueueSequence();
-                defaultSnoozle = queueItem.defaultSnoozle;
-                changeState(INIT);
                 break;
-            case "glucose_after":
-                if (!isWaitingSnoozeResponce) break;
+            case "alarm":
```

Comment:
IMHO, an enum would be more appropriate here and avoids string comparisons at other places as well.